src/resolvers/user.resolver.ts

```diff
@@ -1,17 +1,60 @@
+import { IResolvers } from 'apollo-server';
+import { User } from '../sequelize/models/user.models';
 import UserService from '../services/user.services';
 import { generalValidator } from '../middlewares/general.validator';
-import { loginSchema } from '../utils/schema.utils';
-import { IloginData, Iuser } from '../interfaces/user.interfaces';
-import { generateToken } from '../utils/user.utils';
+import { loginSchema, userSchema } from '../utils/schema.utils';
+import { IloginData, Iuser, Token } from '../interfaces/user.interfaces';
+import { generateToken, returnResult, auth } from '../utils/user.utils';
 
 
 export const userResolver = {
+  Query: {
+    users: (_, { offset = 0, limit = 10 }) => UserService.findAllUsers(offset, limit),
+    user: async (_, { id }) => UserService.findUser(id, 'id'),
+    role: async (_, { id }) => UserService.findRole(id, 'roleQuery'),
+  },
   Mutation: {
-    loginUser: async <T>(_: T, args: IloginData): Promise<any> => {
+    loginUser: async (_: unknown, args: IloginData): Promise<Token> => {
       generalValidator(args, loginSchema);
       const res = await UserService.loginUser(args);
       const token = generateToken(res as Iuser);
       return { token };
     },
+    addUser: async (_, { input }, { token }) => {
+      const decodedUser = await auth(token);
+      generalValidator(input, userSchema);
+      const results = await UserService.createUser(input, decodedUser);
+      return returnResult(results);
+    },
+    updateUser: async (_, { id, input }, { token }): Promise<User> => {
+      const decodedUser = await auth(token);
+      const res = await UserService.updatingUser({ id, input }, decodedUser);
+      return returnResult(res);
+    },
+    deleteUser: async (_, { id }, { token }): Promise<boolean> => {
+      const decodedUser = await auth(token);
+      return !!await UserService.deleteUser(id, decodedUser);
+    },
+    verifyEmail: async (_, { email }, { token }): Promise<unknown> => {
+      const decodedUser = await auth(token);
+      const message = await UserService.emailVerification(email, decodedUser);
+      return { message };
+    },
+  },
+  User: {
+    userRole: async (user, __) => {
+      const res = await UserService.findRole(user.roleId, 'userQuery');
+      return res;
+    },
+    institution: async (user, __) => {
+      const results = await UserService.findUserInstitution(user);
+      return results;
+    },
+  },
+  Role: {
+    users: async (role, __) => {
+      const res = await User.findAll({ where: { roleId: role.id } });
```

Comment:
Let's keep the code related to interaction with the db at the service layer.

Reply:
done